### PR TITLE
[RHV] Add function to reboot RHV VMs

### PR DESCRIPTION
**Pull request by suchita-g**
Signed-off-by: Suchita Gatfane <sgatfane@redhat.com>

### CODE REVIEW

**Comment by suchita-g** [comment]
Marking as verified w.r.t. test Job
https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/3329/
The results have an error for not implemented function in teardown however restart_nodes() functionality executed successfully.

File: ocs_ci/utility/rhv.py [hunk 1]
```diff
@@ -409,3 +407,63 @@ def remove_disks(self, vm, detach_only=True):
                 self.remove_disk(
                     vm=vm, identifier=disk.id, key="id", detach_only=detach_only
                 )
+
+    def get_vm_status(self, vm_name):
+        """
+        Get the power status of RHV VM
+        Args:
```
[the remark this answers is not in the document]
Comment (suchita-g) [reply ocs_ci/utility/rhv.py:414]: Added

File: ocs_ci/utility/rhv.py [hunk 2]
```diff
@@ -409,3 +407,63 @@ def remove_disks(self, vm, detach_only=True):
...
+           vm_name (str): RHV VM name
+        Returns :
```
[the remark this answers is not in the document]
Comment (suchita-g) [reply ocs_ci/utility/rhv.py:417]: Done

File: ocs_ci/utility/rhv.py [hunk 3]
```diff
@@ -409,3 +407,63 @@ def remove_disks(self, vm, detach_only=True):
...
+        vm_service = self.vms_service.vm_service(vm.id)
+        vm_info = vm_service.get()
+        return vm_info.status
+
+    def reboot_rhv_vms(self, vm_names, wait=True, force=True):
+        """
+        Reboot the RHV virtual machines
+        Args:
+            vm_names (list): Names of RHV vms
+            wait (bool): Wait for RHV VMs to reboot
```
[the remark this answers is not in the document]
Comment (suchita-g) [reply ocs_ci/utility/rhv.py:431]: Done

File: ocs_ci/ocs/platform_nodes.py [hunk 4]
```diff
@@ -2186,3 +2186,26 @@ class RHVNodes(NodesBase):
...
+
+    def restart_nodes(self, nodes, timeout=900, wait=True, force=True):
+        """
+        Restart RHV VM
+
+        Args:
+            nodes (list): The OCS objects of the nodes
+            wait (bool): True if need to wait till the restarted node reaches
+                READY state. False otherwise
+            timeout (int): time in seconds to wait for node to reach 'ready' state.
```
Comment (jilju) [inline ocs_ci/ocs/platform_nodes.py:2198]: Please describe *argument* argument

File: ocs_ci/ocs/platform_nodes.py [hunk 5]
```diff
@@ -2186,3 +2186,26 @@ class RHVNodes(NodesBase):
     def __init__(self):
         super(RHVNodes, self).__init__()
         self.rhv = rhv.RHV()
+
+    def restart_nodes(self, nodes, timeout=900, wait=True, force=True):
+        """
+        Restart RHV VM
+
+        Args:
```
Comment (jilju) [inline ocs_ci/ocs/platform_nodes.py:2194]: Please describe *force* argument

File: ocs_ci/utility/rhv.py [hunk 6]
```diff
@@ -487,12 +487,59 @@ def start_rhv_vms(self, vms, wait=True, timeout=600):
...
+                            logger.info(f"RHV Machine {vm.name} reached UP status")
                             break
                 except TimeoutExpiredError:
-                    logger.error(f"RHV VM {vm} is not UP")
+                    logger.error(f"RHV VM {vm.name} is not UP")
+                    raise
+
+    def reboot_rhv_vms(self, vm_names, timeout=600, wait=True, force=True):
+        """
+        Reboot the RHV virtual machines
```
Comment (jilju) [inline ocs_ci/utility/rhv.py:502]: Please add an empty line after 502

**Review by jilju — APPROVED** [review]

File: ocs_ci/ocs/platform_nodes.py [hunk 7]
```diff
@@ -2225,6 +2225,31 @@ def stop_nodes(self, nodes, timeout=600, wait=True, force=True):
...
+            nodes (list): The OCS objects of the nodes
+            timeout (int): time in seconds to wait for node to reach 'ready' state
+            wait (bool): True if need to wait till the restarted node reaches
+                READY state. False otherwise
+            force (bool): True for force VM reboot, False otherwise
+
+        """
+        if not nodes:
+            raise ValueError("No nodes found for restarting")
```
Comment (petr-balogh) [inline ocs_ci/ocs/platform_nodes.py:2241]: This ValueError should be documented in the docstring under `Raises:` section.

Comment (suchita-g) [reply ocs_ci/ocs/platform_nodes.py:2244]: incoperate the review comment

**Review by petr-balogh — APPROVED** [review]

**Review by jilju — APPROVED** [review]

**Pull request merged**